File: Data Structures & Algorithms/valid-sudoku/submission-1.py
class Solution:
    def isValidSudoku(self, arr: List[List[str]]) -> bool:
        rows =collections.defaultdict(set)
        cols =collections.defaultdict(set)
        square =collections.defaultdict(set)
        for i in range(9):
            for j in range(9):
                if arr[i][j]==".":
                    continue
                if (arr[i][j] in rows[i] or arr[i][j] in cols[j] or arr[i][j] in square[(i//3,j//3)]) :
                    return False
                rows[i].add(arr[i][j])
                cols[j].add(arr[i][j])
                square[(i//3,j//3)].add(arr[i][j])
        return True

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& arr, int target) {
  int l = 0, r = arr.size() - 1;
  while (l < r) {
    int resdual = arr[l] + arr[r];
    if (resdual == target) {
      return {l + 1, r + 1};
    }
    if (resdual > target) {
      r--;
    }
    if (resdual < target) {
      l++;
    }
  }
  return {};
    }
};
